src/TensorFlowHub/Utils.cs

```diff
@@ -25,13 +25,25 @@ public static async Task DownloadAsync<TDataSet>(this IModelLoader<TDataSet> mod
             if (!Path.IsPathRooted(dirSaveTo))
                 dirSaveTo = Path.Combine(AppContext.BaseDirectory, dirSaveTo);
 
-            if (!Directory.Exists(dirSaveTo))
-                Directory.CreateDirectory(dirSaveTo);
-            
-            using (var wc = new WebClient())
+            var fileSaveTo = Path.Combine(dirSaveTo, fileName);
+
+            if (File.Exists(fileSaveTo))
             {
-                await wc.DownloadFileTaskAsync(url, Path.Combine(dirSaveTo, fileName));
+                //TODO:maybe you can check file's hashcode and "donglowad.info" to complete file ...
+                Console.WriteLine($"{fileSaveTo} already exists.");
             }
+            else
+            {
+                if (!Directory.Exists(dirSaveTo))
+                    Directory.CreateDirectory(dirSaveTo);
```

Comment:
Directory.CreateDirectory dictates in the documentation: 
> Creates all directories and subdirectories in the specified path unless they already exist.

Reply:
Good to know, I always check at first, :)